### PR TITLE
feat(settings): Allow app discover section links to open app routes or the appstore page

### PR DESCRIPTION
**Pull request by susnux**
* For #42547 

## Summary

This allows `app://{appid}` links to open either the app if installed or the app store page.
Also app routes like `app://files/trashbin` are allowed, those will result in correct routes (e.g. incl. webroot).

https://github.com/nextcloud/server/assets/1855448/5f1d30ca-0807-415a-8e4f-509957016bc4

## TODO

* [ ] Remove the test data commit before merge :exclamation: 

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by szaimen** [comment]
From design perspective I'd say it would be better if the sidebar would open without switching views (keeping the focus on the discover section). Would that be possible?

**Comment by szaimen** [comment]
> @szaimen adjusted it like that. Looks like this:
> 
>  vokoscreenNG-2024-03-18_14-11-15.mp4

Looks great, thanks! :)

**Review by szaimen — APPROVED** [review]
LGTM from design perspective :)

File: apps/settings/lib/Controller/AppSettingsController.php [hunk 1]
```diff
@@ -131,7 +131,12 @@ public function viewApps(): TemplateResponse {
 	 * @NoCSRFRequired
 	 */
 	public function getAppDiscoverJSON(): JSONResponse {
-		$data = $this->discoverFetcher->get();
+		// $data = $this->discoverFetcher->get();
```
Comment (Fenn-CS) [inline apps/settings/lib/Controller/AppSettingsController.php:134]: Reminder to add actual implementation to pull json?

**Review by nfebe — COMMENTED** [review]
Generals looks good, waiting for wrap up?

File: apps/settings/discover.json [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+					"link": "https://example.com/wallpapers"
+				}
+			}
+		}
+	},
+	{
+		"type": "carousel",
+		"id": "1d41a08e-4aa6-49b3-ad1b-ff2e83bcdf96",
+		"headline": {
+			"en": "A carousel",
```
[the remark this answers is not in the document]
Comment (susnux) [reply apps/settings/discover.json:23]: This (commit) will be dropped before merge it is just for testing :)

File: apps/settings/lib/Controller/AppSettingsController.php (hunk 1 above)
Comment (susnux) [reply apps/settings/lib/Controller/AppSettingsController.php:134]: It is implemented just commented out here for testing purpose (last commit will be dropped before merge).

**Review by nfebe — APPROVED** [review]

**Pull request merged**